File: examples/serverless-appsync-python/app/resolvers/handler.py
def getResource(id=None):
    """
    This is the resolver for the getResource field.
    
    id: The id of the resource to get.
    
    Returns: The resource with the given id.
    """

    return {
        'id': id,
        'name': f'Resource {id}'
    }

def getResources():
    """
    This is the resolver for the getResources field.
    
    Returns: The resources.
    """

    return [
        {
            'id': '1',
            'name': 'Resource 1'
        },
        {
            'id': '2',
            'name': 'Resource 2'
        }
    ]

def createResource(name=None):
    """
    This is the resolver for the createResource field.
    
    name: The name of the resource to create.
    
    Returns: The resource that was created.
    """

    return {
        'id': '1',
        'name': name
    }

def graphqlResolver(event, context):
    """
    This is the entry point for the GraphQL API.
    
    event: The event passed to the Lambda function.
    context: The context passed to the Lambda function.
    
    Returns: The response to send back to the client.
    """

    info = event['info']
    arguments = event['arguments']
    fieldName = info['fieldName']

    handlers = {
        'getResource': getResource,
        'getResources': getResources,
        'createResource': createResource
    }

    if fieldName in handlers:
        return handlers[fieldName](**arguments)

    raise Exception('Unknown field name: ' + fieldName)

File: examples/serverless-appsync-node-typescript/src/resolvers/handler.ts
import { AppSyncResolverEvent, Context } from "aws-lambda";

export type AppSyncArguments = {};

const getResource = async ({ id }: { id: string }) => ({
  id,
  name: `Resource ${id}`,
});

const getResources = async () => [
  {
    id: "1",
    name: "Resource 1",
  },
];

const createResource = async ({ name }: { name: string }) => ({
  id: "1",
  name,
});

export const graphqlHandler = async (
  event: AppSyncResolverEvent<AppSyncArguments>,
  context: Context
) => {
  const {
    arguments: eventArguments,
    info: { fieldName },
  } = event;

  const handlers = {
    getResource,
    getResources,
    createResource,
  };

  if (fieldName in handlers) {
    return await handlers[fieldName](eventArguments);
  }

  throw new Error(`Unsupported operation ${fieldName}`);
};
